### PR TITLE
EZP-31825: Use "updateUserPassword" if newer ezplatform-kernel is available

### PR DESCRIPTION
**Pull request by Steveb-p**
| Question      | Answer
| ------------- | ---
| Tickets       |  [EZP-31825](https://jira.ez.no/browse/EZP-31825)
| Bug fix?      | yes
| New feature?  | yes
| BC breaks?    | no
| Tests pass?   | yes/no
| Doc needed?   | no
| License       | [GPL-2.0](https://github.com/ezsystems/ezplatform-admin-ui/blob/master/LICENSE)

Related to https://github.com/ezsystems/ezplatform-kernel/pull/117

#### Checklist:
- [ ] Implement tests
- [ ] Coding standards (`$ composer fix-cs`)

### CODE REVIEW

File: src/bundle/Controller/PasswordChangeController.php [hunk 1]
```diff
@@ -78,7 +78,18 @@ public function userPasswordChangeAction(Request $request)
                 $newPassword = $data->getNewPassword();
                 $userUpdateStruct = $this->userService->newUserUpdateStruct();
                 $userUpdateStruct->password = $newPassword;
-                $this->userService->updateUser($user, $userUpdateStruct);
+                if (method_exists($this->userService, 'updateUserPassword')) {
```
Comment (ViniTou) [inline src/bundle/Controller/PasswordChangeController.php:81]: There is no point of doing it that way, both versions (this and kernel) gonna released at the same time. What you should do is rebase this PR against 1.0 branch.

Comment (alongosz) [reply src/bundle/Controller/PasswordChangeController.php:81]: > There is no point of doing it that way, both versions (this and kernel) gonna released at the same time. What you should do is rebase this PR against 2.1 branch.

@Steveb-p :point_up: bump required ezplatform-kernel version instead

**Review by alongosz — CHANGES_REQUESTED** [review]

File: src/bundle/Controller/PasswordChangeController.php (hunk 1 above)
Comment (Steveb-p) [reply src/bundle/Controller/PasswordChangeController.php:81]: @alongosz @ViniTou what version should be targetted?

Comment (alongosz) [reply src/bundle/Controller/PasswordChangeController.php]: @Steveb-p https://github.com/ezsystems/ezplatform-user/blob/master/composer.json#L21 actually it's already ok - ^1.2@dev, didn't notice

**Review by alongosz — APPROVED** [review]

**Review by ViniTou — APPROVED** [review]

**Review by tomaszszopinski — APPROVED** [review]
QA approved on eZPlatform-ee 3.1.2 with diff & path.

**Pull request merged**